pudb/debugger.py

```diff
@@ -1954,6 +1955,25 @@ def show_output(w, size, key):
             raw_input("Hit Enter to return:")
             self.screen.start()
 
+        def reload_breakpoints_and_redisplay(w, size, key):
+            bp_source_identifier = \
+                self.source_code_provider.get_source_identifier()
+
+            def set_of_line_nums_from_bp_list(bp_list):
+                result = set()
+                for bp in bp_list:
+                    if bp_source_identifier == bp.file and \
+                       bp.line-1 < len(self.source) and bp.enabled:
+                        result.add(bp.line)
+                return result
+            old_bps = set_of_line_nums_from_bp_list(self._get_bp_list())
+            reload_breakpoints(w, size, key)
+            new_bps = set_of_line_nums_from_bp_list(self._get_bp_list())
+            for bp_line in new_bps - old_bps:  # highlight newly added bps
+                self.source[bp_line-1].set_breakpoint(True)
+            for bp_line in old_bps - new_bps:  # de-highlight deleted bps
+                self.source[bp_line-1].set_breakpoint(False)
```

Comment:
Given that this is likely to be infrequent, I would prefer to simply do this by redisplaying the source from scratch:
```
self.set_source_code_provider(..., force_update=True)
```